fs-storage/src/file_storage.rs

```diff
@@ -8,51 +9,110 @@ use std::{
     path::{Path, PathBuf},
 };
 
+use crate::base_storage::BaseStorage;
 use data_error::{ArklibError, Result};
 
 const STORAGE_VERSION: i32 = 2;
 const STORAGE_VERSION_PREFIX: &str = "version ";
 
-pub struct FileStorage {
+pub struct FileStorage<K, V> {
     label: String,
     path: PathBuf,
     timestamp: SystemTime,
+    pub value_by_id: BTreeMap<K, V>,
```

Comment:
Why is `value_by_id` public? 
The API should only expose methods defined in the `BaseStorage` trait. 
For example, we have `as_ref` method that provides access to this field. 

Making this field public would enable users to bypass the methods and access it directly. If users modify this field directly, the `timestamp` wouldn't be updated, potentially leading to corrupt information on our side.